flexget/components/hook/hook.py

```diff
@@ -0,0 +1,336 @@
+                    },
+                },
+            },
+        }
+    )
+
+    @staticmethod
+    def send_hook(title, data: dict, hooks, *args, **kwargs):
+        hooks_run = []
+        for hook in hooks:
+            if Hook.check_run_hook(hook, data, **kwargs):
+                hooks_run.append(hook)
+
+        if not hooks_run:
+            return
+
+        send_hook = plugin.get_plugin_by_name('hook_framework').instance.send_hook
+        send_hook(title, data, hooks, *args, **kwargs)
+
+    @staticmethod
+    def set_config(task: Task, configs: dict):
+        if configs is None:
+            configs = []
+        elif not isinstance(configs, list):
+            configs = [configs]
+
+        for config in configs:
+            events = config.get('events', [])
+            events = events if isinstance(events, list) else [events]
+            config['events'] = events
+
+            stages = config.get('stages', [])
+            stages = stages if isinstance(stages, list) else [stages]
+            config['stages'] = stages
+
+            tasks = config.get('tasks', [])
+            tasks = tasks if isinstance(tasks, list) else [tasks]
+            config['tasks'] = tasks
+
+            plugins = config.get('plugins', [])
+            plugins = plugins if isinstance(plugins, list) else [plugins]
+            config['plugins'] = plugins
+
+            when = config.get('when', [])
+            when = when if isinstance(when, list) else [when]
+            config['when'] = when
+
+            phases = config.get('phases', [])
+            phases = phases if isinstance(phases, list) else [phases]
+            config['phases'] = phases
+
+            via = config.get('via', [])
+            via = via if isinstance(via, list) else [via]
+            config['via'] = via
+
+        Hook.config[task.id] = configs
+
+    @staticmethod
+    def get_config(task: Task):
+        if Hook.config is None or not task.id in Hook.config:
+            return None
+
+        return Hook.config[task.id]
+
+    @staticmethod
+    def delete_config(task: Task):
+        if Hook.config is None or not task.id in Hook.config:
+            return
+
+        del Hook.config[task.id]
+
+    @staticmethod
+    def _is_builtins(plugin_check):
+        return plugin.plugins[plugin_check].builtin
+
+    # First Tasks
+    def on_task_prepare(self, task, config):
+        Hook.set_config(task, config)
+        Hook.on_phase_ended(task, task.current_phase)
+
+    def on_task_start(self, task, config):
+        Hook.set_config(task, config)
+        Hook.on_task_started(task)
+
+    @event('task.execute.started')
+    def on_task_started(self):
+        config = Hook.get_config(self)
+        if not config:
+            return
+
+        Hook.send_hook(
+            None,
+            task_to_dict(self),
+            config,
+            event=ENDPOINT['EVENT_TASK'],
+            stage=ENDPOINT['STAGE_START'],
+            name=getattr(self, 'name'),
```

Comment:
I must admit that I'm not a fan of using the staic here at all, but it was the only way that I was able to make it work, because inside the decorator the self is referring to the task, and not my Hook class!

Reply:
On this case the self is the "task" because its inside the @event... I used getattr to avoid errors in pylint... pylint would assume that the self is the Hook, but in this case the self is the task... so get the task name! Using self.name will also work, but I'll get errors in pylint